yeat/cli/config.py

```diff
@@ -54,3 +56,101 @@ def validate(config):
             keystr = ",".join(sorted(extrakeys))
             message = f"Ignoring unsupported configuration key(s) '{keystr}'"
             warn(message)
+
+    def create_sample_and_assembler_objects(self):
+        self.samples = {}
+        for key, value in self.data["samples"].items():
+            self.samples[key] = Sample.from_data(value)
+        self.assemblers = [
+            Assembler.from_data(assembler, self.threads) for assembler in self.data["assemblers"]
+        ]
+
+    def batch(self):
+        paired_assemblers = []
+        pacbio_assemblers = []
+        for assembler in self.assemblers:
+            if assembler.algorithm in PAIRED:
+                paired_assemblers.append(assembler)
+            elif assembler.algorithm in PACBIO:
+                pacbio_assemblers.append(assembler)
+        self.batch = {
+            "paired": {
+                "samples": self.get_batch_samples(paired_assemblers),
+                "assemblers": paired_assemblers,
+            },
+            "pacbio": {
+                "samples": self.get_batch_samples(pacbio_assemblers),
+                "assemblers": pacbio_assemblers,
+            },
+        }
+
+    def get_batch_samples(self, assemblers):
+        samples = {}
+        for assembler in assemblers:
+            samples = samples | dict(
+                (key, self.samples[key]) for key in assembler.samples if key in self.samples
+            )
+        return samples
+
+    def to_dict(self, args, readtype="all"):
+        if readtype == "all":
+            samples = self.samples
+            assemblers = self.assemblers
+        else:
+            samples = self.batch[readtype]["samples"]
+            assemblers = self.batch[readtype]["assemblers"]
+        return dict(
+            samples={key: value.sample for key, value in samples.items()},
+            labels=[assembler.label for assembler in assemblers],
+            assemblers={assembler.label: assembler.algorithm for assembler in assemblers},
+            extra_args={assembler.label: assembler.extra_args for assembler in assemblers},
+            label_to_samples={assembler.label: assembler.samples for assembler in assemblers},
+            threads=args.threads,
+            downsample=args.downsample,
+            coverage=args.coverage,
+            genomesize=args.genome_size,
+            seed=args.seed,
+        )
+
+
+class Sample:
+    def __init__(self, data):
+        sample = []
+        for fastq in data:
+            read = Path(fastq).resolve()
+            if not read.is_file():
+                raise FileNotFoundError(f"No such file: '{read}'")
+            if str(read) in sample:
+                message = f"Found duplicate read sample: '{read}'"
+                raise AssemblyConfigurationError(message)
+            sample.append(str(read))
+        self.sample = sample
+
+    @classmethod
+    def from_data(cls, data):
+        return cls(data)
```

Comment:
This method isn't...doing anything. There's no difference between calling `Sample(data)` and calling `Sample.from_data(data)`.